### DISCUSSION TITLE
Is it possible to completely disable antialiasing in Bevy?

### DISCUSSION THREAD

**Discussion by micoloth** [bevyengine/bevy#7644] [Q&A] [answered]
As the title says. 

I'm making a very simple 2d sprite-based game, with low res sprites. (kinda like pixel art).

I'd like the sprites to render showing the pixels as is, with no antialiasing.

For reference, this is a zoomed version of the png sprite, where the pixels are clearly visible. 
<img width="221" alt="Screenshot 2023-02-12 at 19 25 32" src="https://user-images.githubusercontent.com/12880257/218329960-9616abf2-1f4d-4420-a318-f4040dba4be2.png">

I'd like this to be the way bevy renders the sprite, even if you zoom the game window.

Instead, you get this:
<img width="42" alt="Screenshot 2023-02-12 at 19 27 28" src="https://user-images.githubusercontent.com/12880257/218329988-dfa0fec8-346c-48bf-bf4e-b776c1c9b41a.png">

which is much uglier..

Can you do 2d rendering with absolutely no antialiasing?

Thanks!

**Comment by alice-i-cecile** [comment, marked as answer]
Yes, see this example :) https://github.com/bevyengine/bevy/blob/main/examples/2d/pixel_perfect.rs

**Reply by alice-i-cecile** [reply]
This is now demonstrated in https://github.com/bevyengine/bevy/blob/41ad4e98de62d247bba8f8f0fec92303ba3a2d4f/examples/2d/sprite_sheet.rs#L8

**Comment by micoloth** [comment]
This is great.. Thanks!

I missed that, but I knew thre had to be a setting somewhere..

Bevy is a very nice project, btw!

**Comment by P-Asta** [comment]
how to use that now?

**Reply by alice-i-cecile** [reply]
`ImagePlugin::default_nearest()` is the incantation you need. Check the link in my reply to myself.

**Reply by P-Asta** [reply]
thx :D

**Comment by P-Asta** [comment]
<img width="1187" alt="image" src="https://github.com/user-attachments/assets/f9832520-405c-4fb1-9a54-ae86060f2b72">
<img width="293" alt="image" src="https://github.com/user-attachments/assets/e12519b0-0a3a-41ba-bead-a0c1e41cc7e7">

```
2024-07-17T23:22:40.355798Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "MacOS 14.4.1 ", kernel: "23.4.0", cpu: "Apple M1", core_count: "8", memory: "16.0 GiB" }
2024-07-17T23:22:40.440744Z  INFO bevy_render::renderer: AdapterInfo { name: "Apple M1", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
thread 'main' panicked at src/main.rs:40:10:
Error adding plugin bevy_app::panic_handler::PanicHandlerPlugin in group bevy_internal::default_plugins::DefaultPlugins: plugin was already added in application
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Reply by jf908** [reply]
It looks like you’re adding DefaultPlugins twice, you only need to add it once.
